ext/exd/provider/github/parse_test.go

```diff
@@ -0,0 +1,147 @@
+		assert.Error(t, actualErr)
+	})
+
+	t.Run("should return nil and error if remote path is not recognized", func(t *testing.T) {
+		remotePath := "invalid-remote-path"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.Nil(t, actualMetadata)
+		assert.ErrorIs(t, actualErr, exd.ErrUnrecognizedRemotePath)
+	})
+
+	t.Run("should return remote metadata with provider github", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor"
+
+		expectedProvider := "github"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Equal(t, expectedProvider, actualMetadata.ProviderName)
+	})
+
+	t.Run("should return remote metadata with the same owner name as expected", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor"
+
+		expectedOwnername := "gojek"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Equal(t, expectedOwnername, actualMetadata.OwnerName)
+	})
+
+	t.Run("should return remote metadata with the same repo name as expected", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor"
+
+		expectedRepoName := "optimus-extension-valor"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Equal(t, expectedRepoName, actualMetadata.RepoName)
+	})
+
+	t.Run("should return remote metadata with empty tag name if not specified", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor"
+
+		expectedTagName := ""
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Equal(t, expectedTagName, actualMetadata.TagName)
+	})
+
+	t.Run("should return remote metadata with the same tag name as expected", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor@v1.0.0"
+
+		expectedTagName := "v1.0.0"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Equal(t, expectedTagName, actualMetadata.TagName)
+	})
+
+	t.Run("should return remote metadata with empty current api path but latest upgrade api path if tag is not specified", func(t *testing.T) {
+		remotePath := "github.com/gojek/optimus-extension-valor"
+
+		expectedUpgradeAPIPath := "https://api.github.com/repos/gojek/optimus-extension-valor/releases/latest"
+
+		actualMetadata, actualErr := github.Parse(remotePath)
+
+		assert.NotNil(t, actualMetadata)
+		assert.NoError(t, actualErr)
+		assert.Empty(t, actualMetadata.CurrentAPIPath)
+		assert.Equal(t, expectedUpgradeAPIPath, actualMetadata.UpgradeAPIPath)
+	})
+
+	t.Run("should return remote metadata with current api path and latest upgrade api path if tag is specified", func(t *testing.T) {
```

Comment:
nit: `should return CurrentAPIPath and UpgradeAPIPath in metadata when tag is specified`